### DISCUSSION TITLE
error validating proposal: access denied: channel [mychannel] creator org unknown, creator is malformed

### DISCUSSION THREAD

**Discussion by sashaswift** [community/community#119649] [Programming Help]
### Select Topic Area

Bug

### Body

The original Identity generated when I deployed the network can run smoothly, but this error occurs when I try to register a new user and run with a new user. This is my SDK document, I would like to ask if there is any problem.

This is my app.js
`"use strict";
/*
 * Copyright IBM Corp. All Rights Reserved.
 *
 * SPDX-License-Identifier: Apache-2.0
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
const grpc = __importStar(require("@grpc/grpc-js"));
const fabric_gateway_1 = require("@hyperledger/fabric-gateway");
const crypto = __importStar(require("crypto"));
const fs_1 = require("fs");
const path = __importStar(require("path"));
const util_1 = require("util");
const AppUtil_1 = require("./utils//AppUtil");
const CAUtil_1 = require("./utils/CAUtil");
const channelName = envOrDefault('CHANNEL_NAME', 'mychannel');
const chaincodeName = envOrDefault('CHAINCODE_NAME', 'sacc');
const mspId = envOrDefault('MSP_ID', 'Org1MSP');
// Path to crypto materials.
const cryptoPath = envOrDefault('CRYPTO_PATH', path.resolve(__dirname, '..', '..', '..', '..', 'hyperledger', 'multinodes', 'crypto-config', 'peerOrganizations', 'org1.example.com'));
// Path to peer tls certificate.
const tlsCertPath = envOrDefault('TLS_CERT_PATH', path.resolve(cryptoPath, 'peers', 'peer0.org1.example.com', 'tls', 'ca.crt'));
// Gateway peer endpoint.
const peerEndpoint = envOrDefault('PEER_ENDPOINT', '124.70.74.28:7051');
const walletPath = path.join(__dirname, 'wallet');
// Gateway peer SSL host name override.
const peerHostAlias = envOrDefault('PEER_HOST_ALIAS', 'peer0.org1.example.com');
const utf8Decoder = new util_1.TextDecoder();
//const assetId = `asset${Date.now()}`;
function envOrDefault(key, defaultValue) {
    return process.env[key] || defaultValue;
}
async function newGrpcConnection() {
    const tlsRootCert = await fs_1.promises.readFile(tlsCertPath);
    const tlsCredentials = grpc.credentials.createSsl(tlsRootCert);
    return new grpc.Client(peerEndpoint, tlsCredentials, {
        'grpc.ssl_target_name_override': peerHostAlias,
    });
}
async function newIdentity(org1UserId) {
    const certDirectoryPath = envOrDefault('CERT_DIRECTORY_PATH', path.resolve(cryptoPath, 'users', org1UserId, 'msp', 'signcerts'));
    const certPath = await getFirstDirFileName(certDirectoryPath);
    const credentials = await fs_1.promises.readFile(certPath);
    return { mspId, credentials };
}
async function getFirstDirFileName(dirPath) {
    const files = await fs_1.promises.readdir(dirPath);
    return path.join(dirPath, files[0]);
}
async function newSigner(org1UserId) {
    const keyDirectoryPath = envOrDefault('KEY_DIRECTORY_PATH', path.resolve(cryptoPath, 'users', org1UserId, 'msp', 'keystore'));
    const keyPath = await getFirstDirFileName(keyDirectoryPath);
    const privateKeyPem = await fs_1.promises.readFile(keyPath);
    const privateKey = crypto.createPrivateKey(privateKeyPem);
    return fabric_gateway_1.signers.newPrivateKeySigner(privateKey);
}
async function registerAndEnroll(org1UserId) {
    await displayInputParameters();
    // The gRPC client connection should be shared by all Gateway connections to this endpoint.
    const ccp = (0, AppUtil_1.buildCCPOrg1)();
    const caClient = (0, CAUtil_1.buildCAClient)(ccp, 'ca.org1.example.com');
    const client = await newGrpcConnection();
    const wallet = await (0, AppUtil_1.buildWallet)(walletPath);
    await (0, CAUtil_1.enrollAdmin)(caClient, wallet, mspId);
    //await enrollAdmin(client, wallet, mspId);
    await (0, CAUtil_1.registerAndEnrollUser)(caClient, wallet, mspId, org1UserId, 'org1.department1');
    // await registerAndEnrollUser(client, wallet,  mspId, org1UserId, 'org1.department1');
    //const client = await newGrpcConnection();
    const identity = await newIdentity(org1UserId);
    const signer = await newSigner(org1UserId);
    const gateway = (0, fabric_gateway_1.connect)({
        client,
        identity,
        signer,
        // Default timeouts for different gRPC calls
        evaluateOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        endorseOptions: () => {
            return { deadline: Date.now() + 15000 }; // 15 seconds
        },
        submitOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        commitStatusOptions: () => {
            return { deadline: Date.now() + 60000 }; // 1 minute
        },
    });
    try {
        // Get a network instance representing the channel where the smart contract is deployed.
        const network = gateway.getNetwork(channelName);
        // Get the smart contract from the network.
        const contract = network.getContract(chaincodeName);
        // Initialize a set of asset data on the ledger using the chaincode 'InitLedger' function.
        console.log('\n--> Evaluate Transaction: ClientAccountID');
        let to = await contract.evaluateTransaction('ClientAccountID');
        let toJson = utf8Decoder.decode(to);
        console.log(`*** Result: ${toJson.toString()}`);
        const user = {
            userIdentity: identity,
            userSigner: signer,
            transactionAccount: toJson.toString()
        };
        gateway.close();
        client.close();
        return JSON.stringify(user);
    }
    catch (error) {
        console.error('******** FAILED to run the application:', error);
        process.exitCode = 1;
        return "error";
    }
}
registerAndEnroll('houfeel8').then(res => {
    console.log(res);
});
async function AppforMint(org1UserId, value) {
    await displayInputParameters();
    // The gRPC client connection should be sIdentityhared by all Gateway connections to this endpoint.
    const client = await newGrpcConnection();
    const identity = await newIdentity(org1UserId);
    const signer = await newSigner(org1UserId);
    const gateway = (0, fabric_gateway_1.connect)({
        client,
        identity,
        signer,
        // Default timeouts for different gRPC calls
        evaluateOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        endorseOptions: () => {
            return { deadline: Date.now() + 15000 }; // 15 seconds
        },
        submitOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        commitStatusOptions: () => {
            return { deadline: Date.now() + 60000 }; // 1 minute
        },
    });
    try {
        // Get a network instance representing the channel where the smart contract is deployed.
        const network = gateway.getNetwork(channelName);
        // Get the smart contract from the network.
        const contract = network.getContract(chaincodeName);
        // Initialize a set of asset data on the ledger using the chaincode 'InitLedger' function.
        console.log('\n--> Submit Transaction: Mint');
        let resultBytes = await contract.submitTransaction('Mint', value);
        const resultJson = utf8Decoder.decode(resultBytes);
        const result = JSON.parse(resultJson);
        console.log(`${JSON.stringify(result)}`);
        gateway.close();
        client.close();
        return (0, AppUtil_1.prettyJSONString)(JSON.stringify(result));
    }
    catch (error) {
        console.error('******** FAILED to run the application:', error);
        process.exitCode = 1;
        return "error";
    }
}
/*AppforMint().catch(error=>{
    console.error('******** FAILED to run the application:', error);
    process.exitCode = 1;
})*/
/**
 * Evaluate a transaction to query ledger state.
 */
async function CreateTransaction(org1UserId, from, id, start, end, to, description) {
    await displayInputParameters();
    // The gRPC client connection should be shared by all Gateway connections to this endpoint.
    const client = await newGrpcConnection();
    const identity = await newIdentity(org1UserId);
    const signer = await newSigner(org1UserId);
    const gateway = (0, fabric_gateway_1.connect)({
        client,
        identity,
        signer,
        // Default timeouts for different gRPC calls
        evaluateOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        endorseOptions: () => {
            return { deadline: Date.now() + 15000 }; // 15 seconds
        },
        submitOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        commitStatusOptions: () => {
            return { deadline: Date.now() + 60000 }; // 1 minute
        },
    });
    try {
        // Get a network instance representing the channel where the smart contract is deployed.
        const network = gateway.getNetwork(channelName);
        // Get the smart contract from the network.
        const contract = network.getContract(chaincodeName);
        // Initialize a set of asset data on the ledger using the chaincode 'InitLedger' function.
        console.log('\n--> Submit Transaction: CreateAsset');
        let resultBytes = await contract.submitTransaction('CreateAsset', from, id, start, end, to, description);
        const resultJson = utf8Decoder.decode(resultBytes);
        const result = JSON.parse(resultJson);
        if (`${JSON.stringify(result)}` !== '') {
            console.log(`*** Result: ${(0, AppUtil_1.prettyJSONString)(JSON.stringify(result))}`);
        }
        gateway.close();
        client.close();
        return (0, AppUtil_1.prettyJSONString)(JSON.stringify(result));
    }
    catch (error) {
        console.error('******** FAILED to run the application:', error);
        process.exitCode = 1;
        return "error";
    }
}
/*CreateTransaction().catch(error=>{
    console.error('******** FAILED to run the application:', error);
    process.exitCode = 1;
})*/
async function UpdateTransaction(org1UserId, id, status, checker) {
    await displayInputParameters();
    // The gRPC client connection should be shared by all Gateway connections to this endpoint.
    const client = await newGrpcConnection();
    const identity = await newIdentity(org1UserId);
    const signer = await newSigner(org1UserId);
    const gateway = (0, fabric_gateway_1.connect)({
        client,
        identity,
        signer,
        // Default timeouts for different gRPC calls
        evaluateOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        endorseOptions: () => {
            return { deadline: Date.now() + 15000 }; // 15 seconds
        },
        submitOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        commitStatusOptions: () => {
            return { deadline: Date.now() + 60000 }; // 1 minute
        },
    });
    try {
        // Get a network instance representing the channel where the smart contract is deployed.
        const network = gateway.getNetwork(channelName);
        // Get the smart contract from the network.
        const contract = network.getContract(chaincodeName);
        // Initialize a set of asset data on the ledger using the chaincode 'InitLedger' function.
        console.log('\n--> Submit Transaction: UpdateAsset');
        let resultBytes = await contract.submitTransaction('UpdateAsset', id, status, checker);
        const resultJson = utf8Decoder.decode(resultBytes);
        const result = JSON.parse(resultJson);
        if (`${JSON.stringify(result)}` !== '') {
            console.log(`*** Result: ${(0, AppUtil_1.prettyJSONString)(JSON.stringify(result))}`);
        }
        gateway.close();
        client.close();
        return (0, AppUtil_1.prettyJSONString)(JSON.stringify(result));
    }
    catch (error) {
        console.error('******** FAILED to run the application:', error);
        process.exitCode = 1;
        return "error";
    }
}
/*UpdateTransaction().catch(error=>{
    console.error('******** FAILED to run the application:', error);
    process.exitCode = 1;
})*/
async function ReadTransaction(org1UserId, id) {
    await displayInputParameters();
    // The gRPC client connection should be shared by all Gateway connections to this endpoint.
    const client = await newGrpcConnection();
    const identity = await newIdentity(org1UserId);
    const signer = await newSigner(org1UserId);
    const gateway = (0, fabric_gateway_1.connect)({
        client,
        identity,
        signer,
        // Default timeouts for different gRPC calls
        evaluateOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        endorseOptions: () => {
            return { deadline: Date.now() + 15000 }; // 15 seconds
        },
        submitOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        commitStatusOptions: () => {
            return { deadline: Date.now() + 60000 }; // 1 minute
        },
    });
    try {
        // Get a network instance representing the channel where the smart contract is deployed.
        const network = gateway.getNetwork(channelName);
        // Get the smart contract from the network.
        const contract = network.getContract(chaincodeName);
        // Initialize a set of asset data on the ledger using the chaincode 'InitLedger' function.
        console.log('\n--> Evaluate Transaction: ReadAsset, function returns "asset1" attributes');
        let resultBytes = await contract.evaluateTransaction('ReadAsset', id);
        const resultJson = utf8Decoder.decode(resultBytes);
        const result = JSON.parse(resultJson);
        if (`${JSON.stringify(result)}` !== '') {
            console.log(`*** Result: ${(0, AppUtil_1.prettyJSONString)(JSON.stringify(result))}`);
        }
        gateway.close();
        client.close();
        return (0, AppUtil_1.prettyJSONString)(JSON.stringify(result));
    }
    catch (error) {
        console.error('******** FAILED to run the application:', error);
        process.exitCode = 1;
        return "error";
    }
}
/*ReadTransaction().catch(error=>{
    console.error('******** FAILED to run the application:', error);
    process.exitCode = 1;
})*/
async function GetAllTransactions(org1UserId) {
    await displayInputParameters();
    // The gRPC client connection should be shared by all Gateway connections to this endpoint.
    const client = await newGrpcConnection();
    const identity = await newIdentity(org1UserId);
    const signer = await newSigner(org1UserId);
    const gateway = (0, fabric_gateway_1.connect)({
        client,
        identity,
        signer,
        // Default timeouts for different gRPC calls
        evaluateOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        endorseOptions: () => {
            return { deadline: Date.now() + 15000 }; // 15 seconds
        },
        submitOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        commitStatusOptions: () => {
            return { deadline: Date.now() + 60000 }; // 1 minute
        },
    });
    try {
        // Get a network instance representing the channel where the smart contract is deployed.
        const network = gateway.getNetwork(channelName);
        // Get the smart contract from the network.
        const contract = network.getContract(chaincodeName);
        // Initialize a set of asset data on the ledger using the chaincode 'InitLedger' function.
        console.log('\n--> Evaluate Transaction: GetAllAssets');
        let resultBytes = await contract.evaluateTransaction('GetAllAssetsForManager');
        const resultJson = utf8Decoder.decode(resultBytes);
        const result = JSON.parse(resultJson);
        if (`${JSON.stringify(result)}` !== '') {
            console.log(`*** Result: ${(0, AppUtil_1.prettyJSONString)(JSON.stringify(result))}`);
        }
        gateway.close();
        client.close();
        return (0, AppUtil_1.prettyJSONString)(JSON.stringify(result));
    }
    catch (error) {
        console.error('******** FAILED to run the application:', error);
        process.exitCode = 1;
        return "error";
    }
}
async function QueryBalance(org1UserId) {
    await displayInputParameters();
    // The gRPC client connection should be shared by all Gateway connections to this endpoint.
    const client = await newGrpcConnection();
    const identity = await newIdentity(org1UserId);
    const signer = await newSigner(org1UserId);
    const gateway = (0, fabric_gateway_1.connect)({
        client,
        identity,
        signer,
        // Default timeouts for different gRPC calls
        evaluateOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        endorseOptions: () => {
            return { deadline: Date.now() + 15000 }; // 15 seconds
        },
        submitOptions: () => {
            return { deadline: Date.now() + 5000 }; // 5 seconds
        },
        commitStatusOptions: () => {
            return { deadline: Date.now() + 60000 }; // 1 minute
        },
    });
    try {
        // Get a network instance representing the channel where the smart contract is deployed.
        const network = gateway.getNetwork(channelName);
        // Get the smart contract from the network.
        const contract = network.getContract(chaincodeName);
        // Initialize a set of asset data on the ledger using the chaincode 'InitLedger' function.
        console.log('\n--> Evaluate Transaction: Balance');
        let resultBytes = await contract.evaluateTransaction('ClientAccountBalance');
        const resultJson = utf8Decoder.decode(resultBytes);
        const result = JSON.parse(resultJson);
        if (`${JSON.stringify(result)}` !== '') {
            console.log(`*** Result: ${(0, AppUtil_1.prettyJSONString)(JSON.stringify(result))}`);
        }
        gateway.close();
        client.close();
        return (0, AppUtil_1.prettyJSONString)(JSON.stringify(result));
    }
    catch (error) {
        console.error('******** FAILED to run the application:', error);
        process.exitCode = 1;
        return "error";
    }
}
/**/
/**
 * Submit a transaction synchronously, blocking until it has been committed to the ledger.
 */
async function displayInputParameters() {
    console.log(`channelName:       ${channelName}`);
    console.log(`chaincodeName:     ${chaincodeName}`);
    console.log(`mspId:             ${mspId}`);
    console.log(`cryptoPath:        ${cryptoPath}`);
    console.log(`tlsCertPath:       ${tlsCertPath}`);
    console.log(`peerEndpoint:      ${peerEndpoint}`);
    console.log(`peerHostAlias:     ${peerHostAlias}`);
}
//# sourceMappingURL=app.js.map`

This is my CAUtil.js
`"use strict";
/*
 * Copyright IBM Corp. All Rights Reserved.
 *
 * SPDX-License-Identifier: Apache-2.0
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.registerAndEnrollUser = exports.enrollAdmin = exports.buildCAClient = void 0;
const fabric_ca_client_1 = __importDefault(require("fabric-ca-client"));
const fs_1 = __importDefault(require("fs"));
const path = __importStar(require("path"));
const cryptoPath = envOrDefault('CRYPTO_PATH', path.resolve(__dirname, '..', '..', '..', '..', '..', 'hyperledger', 'multinodes', 'crypto-config', 'peerOrganizations', 'org1.example.com'));
const usersDirectoryPath = envOrDefault('USERS_DIRECTORY_PATH', path.resolve(cryptoPath, 'users'));
const adminUserId = 'admin';
const adminUserPasswd = 'adminpw';
/**A
 *
 * @param {*} ccp
 */
function envOrDefault(key, defaultValue) {
    return process.env[key] || defaultValue;
}
const buildCAClient = (ccp, caHostName) => {
    // Create a new CA client for interacting with the CA.
    const caInfo = ccp.certificateAuthorities[caHostName]; // lookup CA details from config
    const caTLSCACerts = caInfo.tlsCACerts.pem;
    const caClient = new fabric_ca_client_1.default(caInfo.url, { trustedRoots: caTLSCACerts, verify: false }, caInfo.caName);
    console.log(`Built a CA Client named ${caInfo.caName}`);
    return caClient;
};
exports.buildCAClient = buildCAClient;
const enrollAdmin = async (caClient, wallet, orgMspId) => {
    try {
        // Check to see if we've already enrolled the admin user.
        const identity = await wallet.get(adminUserId);
        if (identity) {
            console.log('An identity for the admin user already exists in the wallet');
            return;
        }
        // Enroll the admin user, and import the new identity into the wallet.
        const enrollment = await caClient.enroll({ enrollmentID: adminUserId, enrollmentSecret: adminUserPasswd });
        const x509Identity = {
            credentials: {
                certificate: enrollment.certificate,
                privateKey: enrollment.key.toBytes(),
            },
            mspId: orgMspId,
            type: 'X.509',
        };
        await wallet.put(adminUserId, x509Identity);
        console.log('Successfully enrolled admin user and imported it into the wallet');
    }
    catch (error) {
        console.error(`Failed to enroll admin user : ${error}`);
    }
};
exports.enrollAdmin = enrollAdmin;
const registerAndEnrollUser = async (caClient, wallet, orgMspId, userId, affiliation) => {
    try {
        // Check to see if we've already enrolled the user
        const userIdentity = await wallet.get(userId);
        if (userIdentity) {
            console.log(`An identity for the user ${userId} already exists in the wallet`);
            return;
        }
        // Must use an admin to register a new user
        const adminIdentity = await wallet.get(adminUserId);
        if (!adminIdentity) {
            console.log('An identity for the admin user does not exist in the wallet');
            console.log('Enroll the admin user before retrying');
            return;
        }
        // build a user object for authenticating with the CA
        const provider = wallet.getProviderRegistry().getProvider(adminIdentity.type);
        const adminUser = await provider.getUserContext(adminIdentity, adminUserId);
        // Register the user, enroll the user, and import the new identity into the wallet.
        // if affiliation is specified by client, the affiliation value must be configured in CA
        const secret = await caClient.register({
            affiliation,
            enrollmentID: userId,
            role: 'client',
        }, adminUser);
        const enrollment = await caClient.enroll({
            enrollmentID: userId,
            enrollmentSecret: secret,
        });
        const x509Identity = {
            credentials: {
                certificate: enrollment.certificate,
                privateKey: enrollment.key.toBytes(),
            },
            mspId: orgMspId,
            type: 'X.509',
        };
        await wallet.put(userId, x509Identity);
        const theUserPath = path.join(usersDirectoryPath, userId);
        fs_1.default.mkdirSync(theUserPath, { recursive: true });
        const mspPath = path.join(theUserPath, 'msp');
        fs_1.default.mkdirSync(mspPath, { recursive: true });
        const mspKeyPath = path.join(mspPath, 'keystore');
        fs_1.default.mkdirSync(mspKeyPath, { recursive: true });
        const mspCertPath = path.join(mspPath, 'signcerts');
        fs_1.default.mkdirSync(mspCertPath, { recursive: true });
        fs_1.default.writeFileSync(path.join(mspCertPath, 'cert.pem'), enrollment.certificate);
        fs_1.default.writeFileSync(path.join(mspKeyPath, 'key.pem'), enrollment.key.toBytes());
        console.log(`User MSP and TLS certificates stored at ${mspPath}`);
        console.log(`Successfully registered and enrolled user ${userId} and imported it into the wallet`);
    }
    catch (error) {
        console.error(`Failed to register user : ${error}`);
    }
};
exports.registerAndEnrollUser = registerAndEnrollUser;
//# sourceMappingURL=CAUtil.js.map`

This is my AppUtil.js
`"use strict";
/*
 * Copyright IBM Corp. All Rights Reserved.
 *
 * SPDX-License-Identifier: Apache-2.0
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.prettyJSONString = exports.buildWallet = exports.buildCCPOrg2 = exports.buildCCPOrg1 = void 0;
const fabric_network_1 = require("fabric-network");
const fs = __importStar(require("fs"));
const path = __importStar(require("path"));
const buildCCPOrg1 = () => {
    // load the common connection configuration file
    const ccpPath = path.resolve(__dirname, '..', '..', '..', '..', 'test-network', 'organizations', 'connection-org1.json');
    const fileExists = fs.existsSync(ccpPath);
    if (!fileExists) {
        throw new Error(`no such file or directory: ${ccpPath}`);
    }
    const contents = fs.readFileSync(ccpPath, 'utf8');
    // build a JSON object from the file contents
    const ccp = JSON.parse(contents);
    console.log(`Loaded the network configuration located at ${ccpPath}`);
    return ccp;
};
exports.buildCCPOrg1 = buildCCPOrg1;
const buildCCPOrg2 = () => {
    // load the common connection configuration file
    const ccpPath = path.resolve(__dirname, '..', '..', '..', '..', 'test-network', 'organizations', 'connection-org2.json');
    const fileExists = fs.existsSync(ccpPath);
    if (!fileExists) {
        throw new Error(`no such file or directory: ${ccpPath}`);
    }
    const contents = fs.readFileSync(ccpPath, 'utf8');
    // build a JSON object from the file contents
    const ccp = JSON.parse(contents);
    console.log(`Loaded the network configuration located at ${ccpPath}`);
    return ccp;
};
exports.buildCCPOrg2 = buildCCPOrg2;
const buildWallet = async (walletPath) => {
    // Create a new  wallet : Note that wallet is for managing identities.
    let wallet;
    if (walletPath) {
        // remove any pre-existing wallet from prior runs
        fs.rmSync(walletPath, { recursive: true, force: true });
        wallet = await fabric_network_1.Wallets.newFileSystemWallet(walletPath);
        console.log(`Built a file system wallet at ${walletPath}`);
    }
    else {
        wallet = await fabric_network_1.Wallets.newInMemoryWallet();
        console.log('Built an in memory wallet');
    }
    return wallet;
};
exports.buildWallet = buildWallet;
const prettyJSONString = (inputString) => {
    if (inputString) {
        return JSON.stringify(JSON.parse(inputString), null, 2);
    }
    else {
        return inputString;
    }
};
exports.prettyJSONString = prettyJSONString;
//# sourceMappingURL=AppUtil.js.map`

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @sashaswift !

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Programming Help category. I've gone ahead and moved it for you. Good luck!
